server/middleware/validationMiddleware.ts

```diff
@@ -1,24 +1,32 @@
 import { plainToInstance } from 'class-transformer'
 import { validate, ValidationError } from 'class-validator'
-import { RequestHandler } from 'express'
-import ConditionService from '../services/conditionService'
+import type { RequestHandler } from 'express'
+import type { Licence } from '../@types/licenceApiClientTypes'
+import type ConditionService from '../services/conditionService'
+import asyncMiddleware from './asyncMiddleware'
 
 export type FieldValidationError = {
   field: string
   message: string
 }
 
+async function getValidator(
+  conditionService: ConditionService,
+  licenceVersion: string | undefined,
+  conditionCode: string,
+  type: new () => object
+): Promise<new () => object> {
+  if (!conditionCode) return type
+  const condition = await conditionService.getAdditionalConditionByCode(conditionCode, licenceVersion)
+  return condition?.validatorType || type
+}
```

Comment:
no short circuiting if no condition code. 

Don't know why the type fall back exists - really horrible as this is used for every POST in the entire application and would need to check every use to see if it actually is used